### PR TITLE
Use getRawStatusCode to avoid getting an exception on fake status code 999, convert it to 406 Not Acceptable

### PR DESCRIPTION
**Pull request by sbrunner**

### CODE REVIEW

**Review by gberaudo — APPROVED** [review]

**Review by sebr72 — CHANGES_REQUESTED** [review]
Please use 406 status code and add test

File: core/src/main/java/org/mapfish/print/map/tiled/CoverageTask.java [hunk 1]
```diff
@@ -282,7 +283,7 @@ private BufferedImage getImageFromResponse(
     }
 
     private Tile handleNonOkStatus(
-        final HttpStatus statusCode, final ClientHttpResponse response, final String baseMetricName)
+        final int statusCode, final ClientHttpResponse response, final String baseMetricName)
```
Comment (sebr72) [inline core/src/main/java/org/mapfish/print/map/tiled/CoverageTask.java:286]: Need to adapt Log messages below

File: core/src/main/java/org/mapfish/print/map/tiled/CoverageTask.java [hunk 2]
```diff
@@ -254,16 +254,17 @@ protected Tile compute() {
...
+        if (statusCode == HttpStatus.NOT_FOUND.value()) {
           LOGGER.info(
               "The request {} returns a not found status code, we consider it as an empty tile.",
               this.tileRequest.getURI());
         }
         // Empty response, nothing special to do
         return new Tile(null, getTileIndexX(), getTileIndexY());
-      } else if (statusCode != HttpStatus.OK) {
+      } else if (statusCode != HttpStatus.OK.value()) {
         return handleNonOkStatus(statusCode, response, baseMetricName);
```
Comment (sebr72) [inline core/src/main/java/org/mapfish/print/map/tiled/CoverageTask.java:268]: Validate status code is redundant then remove it

File: core/src/main/java/org/mapfish/print/map/image/AbstractSingleImageLayer.java [hunk 3]
```diff
@@ -202,14 +202,15 @@ private boolean isResponseStatusCodeValid(
       final String stringBody,
       final String baseMetricName)
       throws IOException {
-    if (httpResponse.getStatusCode() != HttpStatus.OK) {
+    if (httpResponse.getRawStatusCode() != HttpStatus.OK.value()) {
       String message =
           String.format(
-              "Invalid status code for %s (%d!=%d).With request headers:\n%s\n"
+              "Invalid status code for %s (%d!=%d), status: %s. With request headers:\n%s\n"
```
Comment (sebr72) [inline core/src/main/java/org/mapfish/print/map/image/AbstractSingleImageLayer.java:208]: update log %s

**Review by sebr72 — APPROVED** [review]

**Comment by sbrunner** [comment]
Is it ready to be merged?

**Comment by sebr72** [comment]
@sbrunner If you are satisfied with my changes. Please go ahead with the merge.

**Pull request merged**